src/TRestAxionOpticsMirror.cxx

```diff
@@ -0,0 +1,716 @@
+        for (int e = 30; e <= 15000; e += 30) {
+            fHenkeKeys["Fixed"] = IntegerToString(e);
+            cout << ".";
+            cout.flush();
+            string fname = DownloadHenkeFile();
+
+            std::vector<std::vector<Double_t>> data;
+            TRestTools::ReadASCIITable(fname, data, 2);
+
+            // we skip the last point if we are not at the latest angles file
+            Int_t N = data.size() - 1;
+            if (n == 4.5) N = N + 1;
+
+            for (int m = 0; m < N; m++) reflectivity[e].push_back((Float_t)data[m][1]);
+            for (int m = 0; m < N; m++) transmission[e].push_back((Float_t)data[m][2]);
+        }
+        cout << endl;
+    }
+    cout << "Number of angles stored : " << reflectivity[30].size() << endl;
+
+    for (const auto& x : reflectivity) {
+        fReflectivityTable.push_back(x.second);
+    }
+    for (const auto& x : transmission) {
+        fTransmissionTable.push_back(x.second);
+    }
+
+    ExportTables();
+}
+
+///////////////////////////////////////////////
+/// \brief It returns the corresponding reflectivity filename for the mirror properties defined in the data
+/// members
+///
+std::string TRestAxionOpticsMirror::GetReflectivityFilename() {
+    string fnameR = "Reflectivy_" + fMirrorType + "_" + fLayerTop + "_" + fLayerThicknessTop + "_" +
+                    fSubstrate + "_" + fSigmaTop + ".N901f";
+    if (fMirrorType == "Bilayer")
+        fnameR = "Reflectivity_" + fMirrorType + "_" + fLayerTop + "_" + fLayerThicknessTop + "_" +
+                 fSigmaTop + "_" + fLayerBottom + "_" + fLayerThicknessBottom + "_" + fSigmaBottom + "_" +
+                 fSubstrate + ".N901f";
+    return fnameR;
+}
+
+///////////////////////////////////////////////
+/// \brief It returns the corresponding transmission filename for the mirror properties defined in the data
+/// members
+///
+std::string TRestAxionOpticsMirror::GetTransmissionFilename() {
+    string fnameT = "Transmission_" + fMirrorType + "_" + fLayerTop + "_" + fLayerThicknessTop + "_" +
+                    fSubstrate + "_" + fSigmaTop + ".N901f";
+    if (fMirrorType == "Bilayer")
+        fnameT = "Transmission_" + fMirrorType + "_" + fLayerTop + "_" + fLayerThicknessTop + "_" +
+                 fSigmaTop + "_" + fLayerBottom + "_" + fLayerThicknessBottom + "_" + fSigmaBottom + "_" +
+                 fSubstrate + ".N901f";
+    return fnameT;
+}
+
+///////////////////////////////////////////////
+/// \brief It is a private method to export the tables to a binary file once the tables
+/// have been downloaded from Henke database
+///
+Int_t TRestAxionOpticsMirror::ExportTables() {
+    if (fReflectivityTable.size() == 0) {
+        ferr << "Nothing to export!" << endl;
+        return 1;
+    }
+
+    string path = REST_USER_PATH + "/export/";
+
+    if (!TRestTools::fileExists(path)) {
+        cout << "Creating path: " << path << endl;
+        system(("mkdir -p " + path).c_str());
```

Comment:
C++17 finally also has the `std::filesystem` namespace providing `create_directory`:

https://en.cppreference.com/w/cpp/filesystem/create_directory

Reply:
Yes, the problem is that at sultan the ROOT version we are using is compiled with c++14, not sure if there is one available with c++17 @lobis? The docker I believe is c++17